### PR TITLE
CMake: Make dependency paths relocatable

### PR DESCRIPTION
**Pull request by ibauersachs**
The generated configurations for both CMake and pkg-config included absolute paths to dependencies (OpenSSL, MbedTLS). This is contrary to the general CMake advise to create relocatable packages [1].

Additionally, when building both mbedtls and libevent via CMake's FetchContent in the same project, loading the project would fail with

  INTERFACE_INCLUDE_DIRECTORIES property contains path:
    "/home/user/project/cmake-build/_deps/mbedtls-build/include"
  which is prefixed in the source directory.

The required changes include:
- Adding the outer includes only to the BUILD_INTERFACE solves the makes the CMake paths relocatable and thus solves the FetchContent problem.
- Updates to libevent_*.pc.in fixes the relocatable issues for pkg-config and properly declares currently missing dependencies.
- Using components for linking to OpenSSL (requiring CMake 3.4) and MbedTLS. The new MbedTLS target names now match the component names of the MbedTLS' CMake project.
- Use the Threads CMake library reference instead of a direct reference to support both built-in pthread and -lpthread.

[1] https://cmake.org/cmake/help/v3.25/manual/cmake-packages.7.html#creating-relocatable-packages

### CODE REVIEW

**Review by azat — APPROVED** [review]
Thank you! LGTM, though tests failed, let's try to re-run CI.

**Comment by azat** [comment]
This is `thread/priority_inheritance` due to lack of `EVTHREAD_PTHREAD_PRIO_INHERIT`

File: cmake/AddEventLibrary.cmake [hunk 1]
```diff
@@ -11,10 +12,10 @@ macro(set_event_shared_lib_flags LIB_NAME)
...
 
 macro(generate_pkgconfig LIB_NAME)
-    set(prefix      ${CMAKE_INSTALL_PREFIX})
-    set(exec_prefix ${CMAKE_INSTALL_PREFIX})
-    set(libdir      ${CMAKE_INSTALL_LIBDIR})
-    set(includedir  ${CMAKE_INSTALL_INCLUDEDIR})
+    set(prefix      "${CMAKE_INSTALL_PREFIX}")
+    set(exec_prefix "\${prefix}")
+    set(libdir      "\${prefix}/${CMAKE_INSTALL_LIBDIR}")
+    set(includedir  "\${prefix}/${CMAKE_INSTALL_INCLUDEDIR}")
```
Comment (azat) [inline cmake/AddEventLibrary.cmake:18]: @ibauersachs what do you think, maybe here [`CMAKE_INSTALL_FULL_*`](https://cmake.org/cmake/help/v3.0/module/GNUInstallDirs.html) should be used instead?

Comment (ibauersachs) [reply cmake/AddEventLibrary.cmake:18]: No, absolutely not. The whole point of this PR was to remove absolute paths.

**Comment by ibauersachs** [comment]
Thanks for working on this.

Why did you undo some change related to the installation directory, specifically `CMAKE_INSTALL_LIBDIR` to `lib`? Using this variable is essential to properly respect where the user (or another build system) wants the libraries installed.

File: cmake/AddEventLibrary.cmake (hunk 1 above)
Comment (azat) [reply cmake/AddEventLibrary.cmake:18]: Ah, I see, here prefix is an escape, that will be interpreted by pkg-config.

**Comment by azat** [comment]
> Why did you undo some change related to the installation directory, specifically CMAKE_INSTALL_LIBDIR to lib? Using this variable is essential to properly respect where the user (or another build system) wants the libraries installed.

Let me check, AFAIR I've reverted only the change with `CMAKE_REQUIRED_LIBRARIES` (because this breaks `CHECK_SYMBOLS_EXIST`)

So your original patch - https://github.com/libevent/libevent/commit/6e022fead41b9131e353044d44ad4dfd87be557c, patch with my editions - https://github.com/libevent/libevent/commit/acfac7ae4a3edbbb7ce4ceee7208b4245a6e203e

Oh, I see, it had been changed in 81c6b8823c1b58d7837e827bb1098aa5f9e5956b - #1405, where there original patch was correct, so anyway, I simply need to revert 81c6b8823c1b58d7837e827bb1098aa5f9e5956b

**Comment by azat** [comment]
Done - 2c2ffb8d6e27d8e24ca92d407ce950a6402e1c5a

**Comment by azat** [comment]
Thank you for noticing! Also don't hesitate to submit pull request in case you have some thoughts on improving cmake rules!

**Comment by ibauersachs** [comment]
Thanks! I'm not currently working on the project that needed these changes. Also, I'm not a CMake expert, but I'll definitely will always try to upstream local changes.

**Pull request merged**